src/Client/Connection.cpp

```diff
@@ -239,6 +241,7 @@ void Connection::sendHello()
     {
         writeStringBinary(user, *out);
         writeStringBinary(password, *out);
+        writeStringBinary(quota_key, *out);
```

Comment:
This is going to introduce incompatibility between different versions of server and client.

Maybe send quota_key in ClientInfo?

Reply:
Yes, we'll have a problem with different versions. There is an error [here](https://github.com/ClickHouse/ClickHouse/blob/641e2be6ef4fab8b4bb29ffc7482c384894ce13d/src/Server/TCPHandler.cpp#L1034). Client info doesn't have quota_key at this time. Where do you suggest to add it? (it seems to be the first place to get this information)

Reply:
Check the server's revision and serialize quota_key only for supporting servers.